pandas/core/groupby/ops.py

```diff
@@ -881,14 +888,14 @@ def fast_apply(self, f, names):
         sdata = self._get_sorted_data()
         return libreduction.apply_frame_axis0(sdata, f, names, starts, ends)
 
-    def _chop(self, sdata, slice_obj: slice):
+    def _chop(self, sdata: DataFrame, slice_obj: slice) -> DataFrame:
```

Comment:
should this be Series?

Reply:
im pretty sure this is correct as-is

Reply:
ahh ok, had the class FrameSplitter hidden and just saw the class SeriesSplitter above. I guess sdata doesn't mean series data then.